src/Result.test.ts

```diff
@@ -1,177 +1,248 @@
-			expect(result.childResults).toEqual([]);
-			expect(result).toMatchSnapshot();
-		});
+			it("should create a Result with issues from Issue objects", () => {
+				const issues = [{ message: "issue 1" }, { message: "issue 2" }];
+				const result = new Result(issues);
 
-		it("should create a Result with issues from Issue objects", () => {
-			const issues = [{ message: "issue 1" }, { message: "issue 2" }];
-			const result = createValidationResult(issues);
+				expect(result.issues).toEqual(issues);
+				expect(result.errorMessages).toEqual(
+					issues.map((issue) => issue.message),
+				);
+				expect(result.childResults).toEqual([]);
+			});
 
-			expect(result.issues).toEqual(issues);
-			expect(result.errorMessages).toEqual(
-				issues.map((issue) => issue.message),
-			);
-			expect(result.childResults).toEqual([]);
-			expect(result).toMatchSnapshot();
+			it("should create a Result with child results", () => {
+				const issues = [{ message: "issue 1" }, { message: "issue 2" }];
+				const childResults = [
+					new ChildResult(0, ["child issue 1"]),
+					new ChildResult(1, ["child issue 2"]),
+				];
+				const result = new Result(issues, childResults);
+
+				expect(result.issues).toEqual(issues);
+				expect(result.errorMessages).toEqual([
+					...issues.map((issue) => issue.message),
+					...childResults[0].errorMessages,
+					...childResults[1].errorMessages,
+				]);
+				expect(result.childResults).toEqual(childResults);
+			});
 		});
 
-		it("should create a Result with child results", () => {
-			const issues = [{ message: "issue 1" }, { message: "issue 2" }];
-			const childResults = [
-				{ ...createValidationResult(["child issue 1"]), index: 0 },
-				{ ...createValidationResult(["child issue 2"]), index: 1 },
-			];
-			const result = createValidationResult(issues, childResults);
-
-			expect(result.issues).toEqual(issues);
-			expect(result.errorMessages).toEqual([
-				...issues.map((issue) => issue.message),
-				...childResults[0].errorMessages,
-				...childResults[1].errorMessages,
-			]);
-			expect(result.childResults).toEqual(childResults);
-			expect(result).toMatchSnapshot();
-		});
-	});
-	describe("addChildResult", () => {
-		it("should add a child Result to a parent Result with a Result object", () => {
-			const parent = createValidationResult();
-			const child = createValidationResult(["child issue"]);
+		describe("addChildResult", () => {
+			it("should add a child Result with a Result object", () => {
+				const parent = new Result();
+				const child = new Result(["child issue"]);
 
-			addChildResult(parent, child, 0);
+				parent.addChildResult(0, child);
 
-			expect(parent.childResults).toHaveLength(1);
-			expect(parent.childResults[0]).toEqual({ ...child, index: 0 });
-			expect(parent.errorMessages).toEqual(child.errorMessages);
-		});
+				expect(parent.childResults).toHaveLength(1);
+				expect(parent.childResults[0]).toEqual(new ChildResult(0, child));
```

Comment:
[Testing] I know this is existing code and totally not at blocker level, but maybe this would be a little cleaner?

```suggestion
				expect(parent.childResults).toEqual([new ChildResult(0, child)]);
```

(here and elsewhere)

Reply:
Updated